server/reminders/reminderApi.ts

```diff
@@ -0,0 +1,137 @@
+import crypto from 'crypto';
+import { captureMessage } from '@sentry/node';
+import type { Request, Response } from 'express';
+import fetch from 'node-fetch';
+import { s3ConfigPromise } from '../awsIntegration';
+import { conf } from '../config';
+import type { ReminderType } from './reminderData';
+import { addReminderPeriod } from './reminderData';
+
+interface RemindersConfig {
+	reminderHmacKey: string;
+}
+
+const reminderConfigPromise = s3ConfigPromise<RemindersConfig>()('reminders');
+
+const getReminderHmacKey = (): Promise<string> =>
+	reminderConfigPromise.then((result) => {
+		if (result?.reminderHmacKey) {
+			return result.reminderHmacKey;
+		} else {
+			return Promise.reject('Failed to get reminder Hmac key');
+		}
+	});
+
+const verifyReminderToken = (
+	reminderData: string,
+	token: string,
+	key: string,
+): boolean => {
+	const hash = crypto
+		.createHmac('sha1', key)
+		.update(reminderData)
+		.digest('hex');
+
+	return hash === token;
+};
+
+export const createOneOffReminderHandler = (req: Request, res: Response) =>
+	createReminder('ONE_OFF', req.body).then((response) => {
+		if (!response.ok) {
+			captureMessage('Reminder sign up failed at the point of request');
+		}
+		res.sendStatus(response.status);
+	});
+
+// Instead of requiring the user to be signed in, this endpoint verifies the token
+export const publicCreateReminderHandler =
+	(reminderType: ReminderType) => async (req: Request, res: Response) => {
+		const { reminderData, token } = JSON.parse(req.body);
+
+		if (reminderData && token) {
+			const reminderHmacKey = await getReminderHmacKey();
```

Comment:
Is there a chance this will fail (if we can't get the key)? Do we need to catch and return an explicit error if that happens?

Reply:
I've refactored the error handling in this function. It's tricky to make it elegant but I think it's a bit better now